File: single-language-analysis/dataset-files/python/test_smell_095.py
# Original URL: https://github.com/ibmdbanalytics/ibmdbpy/blob/c7f1cd3892b609644e3d68fcbdec151ac4634612/ibmdbpy/tests/test_frame.py#L315-L331

class Test_DataExploration(object):

    def test_smell_095(self, idadf, df):
        if len(idadf.columns) >= 4:
            columns = idadf.columns[1:4].tolist()
            newidadf = idadf[columns]

            sortkey = newidadf.columns[0]
            if newidadf._get_numerical_columns():
                sortkey = newidadf._get_numerical_columns()[0]

            ida_head = newidadf.head()

            df_sorted = df.sort_values(sortkey)
            df_head = df_sorted[columns].head()

            assert isinstance(ida_head, pandas.core.frame.DataFrame)
            assert len(ida_head) == 5
            assert(ida_head[sortkey].tolist() == df_head[sortkey].tolist())

File: single-language-analysis/dataset-files/java/TestSmell095.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.util.List;

public class TestSmell095 {

    @Test
    public void testIdadfHeadProjected3col() {
        if (idadf.getColumns().length >= 4) {
            String[] columns = new String[]{idadf.getColumns()[1], idadf.getColumns()[2], idadf.getColumns()[3]};
            DataFrame newIdadf = idadf.get(columns);

            String sortKey = newIdadf.getColumns()[0];
            if (newIdadf.getNumericalColumns().length > 0) {
                sortKey = newIdadf.getNumericalColumns()[0];
            }

            DataFrame idaHead = newIdadf.head();

            DataFrame dfSorted = df.sortValues(sortKey);
            DataFrame dfHead = dfSorted.get(columns).head();

            assertTrue(idaHead instanceof DataFrame);
            assertEquals(5, idaHead.length());
            assertArrayEquals(dfHead.get(sortKey).toArray(), idaHead.get(sortKey).toArray());
        }
    }
}